src/logics/LRALogic.cc

```diff
@@ -50,90 +51,26 @@ const char* LRALogic::tk_real_geq   = ">=";
 const char* LRALogic::s_sort_real = "Real";
 
 LRALogic::LRALogic() :
-    LALogic()
-        , sym_Real_ZERO(SymRef_Undef)
-        , sym_Real_ONE(SymRef_Undef)
-        , sym_Real_NEG(SymRef_Undef)
-        , sym_Real_MINUS(SymRef_Undef)
-        , sym_Real_PLUS(SymRef_Undef)
-        , sym_Real_TIMES(SymRef_Undef)
-        , sym_Real_DIV(SymRef_Undef)
-        , sym_Real_EQ(SymRef_Undef)
-        , sym_Real_LEQ(SymRef_Undef)
-        , sym_Real_LT(SymRef_Undef)
-        , sym_Real_GEQ(SymRef_Undef)
-        , sym_Real_GT(SymRef_Undef)
-        , sym_Real_ITE(SymRef_Undef)
-        , sort_REAL(SRef_Undef)
-        , term_Real_ZERO(PTRef_Undef)
-        , term_Real_ONE(PTRef_Undef)
-        , term_Real_MINUSONE(PTRef_Undef)
+          LALogic()
+        , sort_REAL(declareSortAndCreateFunctions(s_sort_real))
+        , term_Real_ZERO(mkConst(sort_REAL, tk_real_zero))
+        , term_Real_ONE(mkConst(sort_REAL, tk_real_one))
+        , term_Real_MINUSONE(mkConst(sort_REAL, tk_real_minusone))
+        , sym_Real_ZERO(getSymRef(term_Real_ZERO))
+        , sym_Real_ONE(getSymRef(term_Real_ONE))
+        , sym_Real_NEG(declareFun_NoScoping(tk_real_neg, sort_REAL, {sort_REAL}))
+        , sym_Real_MINUS(declareFun_NoScoping_LeftAssoc(tk_real_minus, sort_REAL, {sort_REAL, sort_REAL}))
+        , sym_Real_PLUS(declareFun_Commutative_NoScoping_LeftAssoc(tk_real_plus, sort_REAL, {sort_REAL, sort_REAL}))
+        , sym_Real_TIMES(declareFun_Commutative_NoScoping_LeftAssoc(tk_real_times, sort_REAL, {sort_REAL, sort_REAL}))
+        , sym_Real_DIV(declareFun_NoScoping_LeftAssoc(tk_real_div, sort_REAL, {sort_REAL, sort_REAL}))
+        , sym_Real_EQ(declareFun_Commutative_NoScoping_Chainable(tk_equals, sort_BOOL, {sort_REAL, sort_REAL}))
+        , sym_Real_LEQ(declareFun_NoScoping_Chainable(tk_real_leq, sort_BOOL, {sort_REAL, sort_REAL}))
+        , sym_Real_LT(declareFun_NoScoping_Chainable(tk_real_lt, sort_BOOL, {sort_REAL, sort_REAL}))
+        , sym_Real_GEQ(declareFun_NoScoping_Chainable(tk_real_geq, sort_BOOL, {sort_REAL, sort_REAL}))
+        , sym_Real_GT(declareFun_NoScoping_Chainable(tk_real_gt, sort_BOOL, {sort_REAL, sort_REAL}))
+        , sym_Real_ITE(declareFun(tk_ite, sort_REAL, {sort_BOOL, sort_REAL, sort_REAL}))
```

Comment:
And here `ITE` does not even get `no scoping` attribute.
What is the difference?

Reply:
I refactored this code.  Now the core theory functions are all defined in `declareSortAndCreateFunctions`, and the symbol names are read & stored from the hash tables that the function updates.

The only exception should be for the Boolean logic that is handled differently for bootstrapping reasons.